src/map.cpp

```diff
@@ -602,59 +602,49 @@ float map::vehicle_vehicle_collision( vehicle &veh, vehicle &veh2,
-        int &x_cof1 = cof1.x;
-        int &y_cof1 = cof1.y;
-        int &x_cof2 = cof2.x;
-        int &y_cof2 = cof2.y;
-        rl_vec2d collision_axis_y;
-
-        collision_axis_y.x = ( veh.global_pos3().x + x_cof1 ) - ( veh2.global_pos3().x + x_cof2 );
-        collision_axis_y.y = ( veh.global_pos3().y + y_cof1 ) - ( veh2.global_pos3().y + y_cof2 );
-        collision_axis_y = collision_axis_y.normalized();
-        rl_vec2d collision_axis_x = collision_axis_y.rotated( M_PI / 2 );
-        // imp? & delta? & final? reworked:
-        // newvel1 =( vel1 * ( mass1 - mass2 ) + ( 2 * mass2 * vel2 ) ) / ( mass1 + mass2 )
+        float E_before = 0.5 * m1 * velocity_veh1.magnitude() * velocity_veh1.magnitude() +
+                         0.5 * m2 * velocity_veh2.magnitude() * velocity_veh2.magnitude();
+
+
         // as per http://en.wikipedia.org/wiki/Elastic_collision
-        //velocity of veh1 before collision in the direction of collision_axis_y
-        float vel1_y = collision_axis_y.dot_product( velo_veh1 );
-        float vel1_x = collision_axis_x.dot_product( velo_veh1 );
-        //velocity of veh2 before collision in the direction of collision_axis_y
-        float vel2_y = collision_axis_y.dot_product( velo_veh2 );
-        float vel2_x = collision_axis_x.dot_product( velo_veh2 );
-        // e = 0 -> inelastic collision
-        // e = 1 -> elastic collision
-        float e = get_collision_factor( vel1_y / 100 - vel2_y / 100 );
-
-        // Velocity after collision
-        // vel1_x_a = vel1_x, because in x-direction we have no transmission of force
-        float vel1_x_a = vel1_x;
-        float vel2_x_a = vel2_x;
-        // Transmission of force only in direction of collision_axix_y
-        // Equation: partially elastic collision
-        float vel1_y_a = ( m2 * vel2_y * ( 1 + e ) + vel1_y * ( m1 - m2 * e ) ) / ( m1 + m2 );
-        float vel2_y_a = ( m1 * vel1_y * ( 1 + e ) + vel2_y * ( m2 - m1 * e ) ) / ( m1 + m2 );
-        // Add both components; Note: collision_axis is normalized
-        rl_vec2d final1 = collision_axis_y * vel1_y_a + collision_axis_x * vel1_x_a;
-        rl_vec2d final2 = collision_axis_y * vel2_y_a + collision_axis_x * vel2_x_a;
-
-        veh.move.init( final1.x, final1.y );
-        veh.velocity = final1.magnitude();
-
-        veh2.move.init( final2.x, final2.y );
-        veh2.velocity = final2.magnitude();
-        //give veh2 the initiative to proceed next before veh1
+        // There is no solution for the 2D case where vehicle 1 and vehicle 2 both are moving at some arbitrary angle
+        // We simplify the situation to 1D case along the vehicle 1 velocity vector
+        // If vehicle 2 is stationary or moving along the same vector as vehicle 1 the result is 100% right
+        // If vehicle 2 is moving fast in a completely different direction the result may be a bit odd
+
+        const float component_velocity_veh1 = velocity_veh1.magnitude();
```

Comment:
```suggestion
        const float comp_velo_veh1 = velo_veh1.magnitude();
```
along with a note that this is the component of vehicle 1's velocity vector along the axis of vehicle 1's velocity vector.

These fairly long variable names become unwieldy immediately.  For instance, replacing:
```
        float vel1_y_a = ( m2 * vel2_y * ( 1 + e ) + vel1_y * ( m1 - m2 * e ) ) / ( m1 + m2 );
```
with
```
        const float component_velocity_veh1_after = ( elasticity * m2 * ( component_velocity_veh2 -
￼          		￼                 component_velocity_veh1 ) + m1 * component_velocity_veh1 + m2 * component_velocity_veh2 ) /
￼         		￼                 ( m1 + m2 );
```
does not really improve readability in my mind.